include/open62541pp/types.hpp

```diff
@@ -888,59 +903,87 @@ class Variant : public TypeWrapper<UA_Variant, UA_TYPES_VARIANT> {
 public:
     using TypeWrapper::TypeWrapper;  // inherit constructors
 
+    // Universal constructors with default copy policy
+
+    template <typename ConstructorToken = ConstructorToken<VariantPolicy::Copy>, typename T>
+    Variant(T&& value, ConstructorToken /*unused*/) {
+        setValue<ConstructorToken::policy>(std::forward<T>(value));
+    }
+
+    template <typename ConstructorToken = ConstructorToken<VariantPolicy::Copy>, typename T>
+    Variant(T&& value, const UA_DataType& dataType, ConstructorToken /*unused*/) {
+        setValue<ConstructorToken::policy>(std::forward<T>(value), dataType);
+    }
+
+    template <typename ConstructorToken = ConstructorToken<VariantPolicy::Copy>, typename InputIt>
+    Variant(InputIt first, InputIt last, ConstructorToken /*unused*/) {
+        setValue<ConstructorToken::policy>(first, last);
+    }
+
+    template <typename ConstructorToken = ConstructorToken<VariantPolicy::Copy>, typename InputIt>
+    Variant(InputIt first, InputIt last, const UA_DataType& dataType, ConstructorToken /*unused*/) {
+        setValue<ConstructorToken::policy>(first, last, dataType);
+    }
+
+    // TODO:
+    // Do not add fromValue methods? As we the constructors above.
```

Comment:
Which variant do you prefer @lukasberbuer ?

- Only constructors,
- Only `fromValue` or
- Both constructors and `fromValue?`

Reply:
The `Variant` interface is already quite complex. I would prefer constructors only.

Reply:
Removed the `fromValue` functions